ext/libxml/ruby_xml_node.c

```diff
@@ -45,13 +45,18 @@ static void rxml_node_deregisterNode(xmlNodePtr xnode)
   /* Has the node been wrapped and exposed to Ruby? */
   if (xnode->_private)
   {
+    /* Does it belong to another libxml user? */
+    if (!OWNED(xnode))
```

Comment:
Agreed.

To clarify my thinking...

First I would refactor to a single-condition and remove a level of nesting:

```c
if (xnode->_private && OWNED(xnode)) {
 // do stuff
```

Then I would return early to remove the block:

```
```c
if (!xnode->_private || !OWNED(xnode))
 return;
```

But I _think_ (`!xnode->_private || !OWNED(xnode)`) is a tautology (or at least `!OWNED(xnode) || !xnode->_private` is)... so I would remove the redundancy:

```c
if (!OWNED(xnode))
 return;
```